src/webots/nodes/utils/WbSupervisorUtilities.cpp

```diff
@@ -1071,6 +1092,42 @@ void WbSupervisorUtilities::handleMessage(QDataStream &stream) {
             mFoundFieldId = id;
             mFoundFieldType = field->type();
             mFoundFieldIsInternal = allowSearchInProto == 1;
+            mFieldName = field->name();
+            std::cout << field->name().toUtf8().constData() << std::endl << std::flush;
+            if (mv) {
+              mWatchedFields.append(WbUpdatedFieldInfo(node->uniqueId(), field->name(), mFoundFieldCount));
+              field->listenToValueSizeChanges();
+              connect(field, &WbField::valueSizeChanged, this, &WbSupervisorUtilities::notifyFieldUpdate, Qt::UniqueConnection);
+            }
+          }
+        }
+      }
+      return;
+    }
+    case C_SUPERVISOR_FIELD_GET_FROM_ID: {
+      int node_id, field_id;
```

Comment:
Nodes are searched using a unique id, but fields don't have ids but only an index in the nodes fields list.
```suggestion
    case C_SUPERVISOR_FIELD_GET_FROM_INDEX: {
```